tests/unit/src/test/resources/inlayHints3/example/GivenAlias.scala

```diff
@@ -28,7 +28,7 @@ trait Zg[T]:
 given Xg with
   def doX/*: Int<<scala/Int#>>*/ = 7
 
-given (using Xg): Yg with
+given (using /*x$1<<(30:13)>>: */Xg): Yg with
```

Comment:
If I understand correctly, in these anonymous givens we will add compiler generated names, but is this really helpful to the end user? I don't see any use case in which this can help. If someone wants to use it as value later in code, it should rather be named implicit param. 

I'd at least add option to hide those decorations as I see them as unnecessary noise.

Reply:
Maybe we could rename the synthetic compiler generated names to something like `<anonymous>`? :thinking: The nice thing about these hints is that they provide a "go to symbol" functionality, which can still be helpful. :slightly_smiling_face:

Reply:
I would keep `x$1` since this an actual value that you can use. So best to be consistent with the compiler

Reply:
I don't think it's necessary and it would harder to see where it's actually defined.

Reply:
What I mean. If we show it, we may suggest to someone to use it, and this is only visible via decorations. Imagine scenario when another user doesn't have decorations enabled in the editor and the confusion he must see. I still think that in this scenario we should not show this.

When we apply this method its a whole different scenario. This will allow us to navigate to the actual implicit and may be useful, but at this point the user should summon it explicitly.

Reply:
So I stand at my position: leave it for applications, remove it for definitions.

Reply:
TBH I think it's highly unlikely anybody would use the anonymous `x$1` in the code without giving it a name first. On the other hand, wanting to see where the implicit value is coming from and going there is something that's used and desired quire often I would say. :wink:

Reply:
Yeah, but where would you want to go in this definition ?

```scala
def test(using /*x$1*/: Int) = ???
```
If you try to go to definition of `x$1`, this is the definition.

```scala
def test(using <</*x$1*/>> Int) =
  test(/*using x$1*/) // we can go to definition from there to << >>
given Int = 5
test(/*using given_Int*/) // and there
```
But if there is no x$1 in param name, we can also go with cursor to `Int`
```scala
def test(using <<Int>>) =
  test(/*using x$1*/) // we can go to definition from there
given Int = 5
test(/*using given_Int*/) // and there
```

Reply:
I think I see what you mean, thanks. :+1: As long as we can get a hint that takes us from the application to the correct implicit definition I think we are golden, sorry for the misunderstanding. :sweat_smile:

Reply:
It is always good to discuss stuff and brainstorm ideas 😆

Reply:
```scala
def test(using <<Int>>) =
  test(/*using x@@$1*/)
```

Reply:
The code won't compile if there is ambiguity, so the identifier is not necessary

Reply:
But if there is named implicit in scope it should be like this:
```scala
def test(using test: <<Int>>) =
  test(/*using test: In@@t*/)
``` or even shorter


```scala
def test(using <<test>>: Int) =
  test(/*using te@@st*/)
```

Reply:
` test(/*using In@@t*/)` is not a valid code though, it would be weird to show that

Reply:
The problem is in keeping those decorations minimal. There will be tons of them, and in my opinion we should focus on displaying actually useful information. This may work, but even in simple projects, it will be added to almost every method to e.g. pass execution context which most users won't name. As long as we can go from usage to source, it's fine. The fact that decoration is valid Scala code is not as important as it is to show useful info.

Reply:
Coming back to it, you are assuming that some information will not be useful. And it might not be useful to you, but might be useful to someone else. Having the added name at definition fits very well with all the other decorations, where we add them in places that the compiler will add something itself. We also have options to turn off certain decorations, to toggle them etc. So if there is too much at any point, they can be easily hidden. Also this is not really an issue in cases that decorations don't show all together.

Reply:
I still don't agree that this can be useful to anyone. Compiler will report error on ambiguity, and you can still find the source, I don't think anything else is necessary. 

If we go with this logic there are tons of other scenarios in which we can show such decorations, even the context bounds so we also should adjust them to display where is this name coming from: 
```scala
def test[T: Ordering](x: List[T])/*(using $evidence1: Ordering[T])*/ =
  x.sorted/*(using $evidence1)*/
```
Which I think right now would lead to the same issue of not knowing where `$evidence1` comes from
```scala
def test[T: Ordering](x: List[T]) =
  x.sorted/*(using $evidence1)*/
```

and even in this simple example it will already look cluttered:
```scala
def test[T: Ordering](x: List[T])/*(using $evidence1: Ordering[T])*/ =
  test/*[T]*/(x)/*(using $evidence1)*/
``` 

so maybe another representation may help with this ?

```scala
def test[T: Ordering](x: List[T]) =
  x.sorted/*(using Ordering)*/
```

and how would it look in the scenario:
```scala
def test[T: Ordering](x: List[T]) =
  x.sorted
trait Foo
given Ordering[Foo] = ???
test(List.empty[Foo])
```
would it be:
```scala
def test[T: Ordering](x: List[T])/*(using $evidence1: Ordering[T])*/ =
  x.sorted/*(using $evidence1)*/
trait Foo
given Ordering[Foo] = ???
test/*[Foo]*/(List.empty[Foo])/*(using given_Ordering_Foo)*/
```
? Maybe we should also consider something shorter:
```scala
test/*[Foo: Ordering]*/(List.empty[Foo])
```
and trying to go to definiton on `Ordering` would put you in proper definition ?

Nevertheless, these are just the ideas. If we add an option to configure it at that level, I'm fine with this change, but afaik at the moment we can only control to hide / display implicits all together. We could make it available via `verbose` setting and leave it normal with `on`, or even more customisation options, including other decorations.

Reply:
I think that because users can precisely decide which decorations they find useful, there is no reason to remove this feature (inlay hints in anonymous given definitions), let's just hide it under a separate option.

> Nevertheless, these are just the ideas. If we add an option to configure it at that level, I'm fine with this change, but afaik at the moment we can only control to hide / display implicits all together. We could make it available via verbose setting and leave it normal with on, or even more customisation options, including other decorations.

I am fine with this, in this PR we can only add `using` to implicit parameters and in a follow up decouple decorations for context bounds parameters and add decorations in definitions, both under separate options.